File: Blind75/217-contain-duplicate.py
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        table = {} # 1. set a new empty dictionary
        for i in nums: # 2. run through the array
            if i in table: # 3. if the number is already in the dictionary, return True
                return True
            else:
                table[i] = 1 # 4. else, add the number to the dictionary
        return False

File: Blind75/217-contain-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */

// use hash table
var containsDuplicate = function(nums) {
  let map = new Map();
  for (let i = 0; i < nums.length; i++){
      if (map.has(nums[i])) {
          return true
      }
      map.set(nums[i], 1)
  }
  return false
};
